src/lib/proof-system/zkprogram.ts

```diff
@@ -265,10 +264,26 @@ function ZkProgram<
   [I in keyof Config['methods']]: Prover<
     InferProvableOrUndefined<Get<Config, 'publicInput'>>,
     InferProvableOrVoid<Get<Config, 'publicOutput'>>,
-    PrivateInputs[I],
-    InferProvableOrUndefined<AuxiliaryOutputs[I]>
+    InferPrivateInput<Config>[I],
+    InferProvableOrUndefined<InferAuxilaryOutputs<Config>[I]>
   >;
 } {
+  type Methods = {
+    [I in keyof Config['methods']]: Method<
+      InferProvableOrUndefined<Get<Config, 'publicInput'>>,
+      InferProvableOrVoid<Get<Config, 'publicOutput'>>,
+      Config['methods'][I]
+    >;
+  };
+  // derived types for convenience
+  type MethodSignatures = Config['methods'];
+  type PrivateInputs = {
+    [I in keyof Config['methods']]: Config['methods'][I]['privateInputs'];
+  };
+  type AuxiliaryOutputs = {
+    [I in keyof MethodSignatures]: Get<MethodSignatures[I], 'auxiliaryOutput'>;
+  };
```

Comment:
you use the derived types above here as well?

Reply:
You're right, i've fixed it to also use the global derived types here